compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -492,6 +499,32 @@ impl Inliner<'tcx> {
                     }
                 }
 
+                if self.insert_storage_markers {
+                    for local in callee_body.vars_and_temps_iter() {
+                        if integrator.always_live_locals.contains(local) {
+                            let new_local = integrator.map_local(local);
+                            caller_body[callsite.block].statements.push(Statement {
+                                source_info: callsite.source_info,
+                                kind: StatementKind::StorageLive(new_local),
+                            });
+                        }
+                    }
+                    if let Some(block) = callsite.target {
+                        let mut n = 0;
```

Comment:
Leave a comment that these statements are added in the beginning of the block by adding them to the end and then rotating them to the front.

Alternatively use the [mir patch system](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_mir/util/patch/struct.MirPatch.html).